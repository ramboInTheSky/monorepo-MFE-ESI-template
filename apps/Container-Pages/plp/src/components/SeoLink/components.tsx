import styled from "styled-components"

export const SEOLinkComponent = styled.a`
    margin-top: 2rem;
    padding: 1rem 4rem;
    border: 1px solid black;
    color: black;
    display: block;
    list-style: none;
`
