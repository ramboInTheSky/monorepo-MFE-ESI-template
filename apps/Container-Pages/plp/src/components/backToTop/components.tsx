import styled from "styled-components"
import Zoom from "@mui/material/Zoom"
import Fab from "@mui/material/Fab"

export const StyledZoom = styled(Zoom)`
    transition-delay: 0.3s;
`

export const StyledFab = styled(Fab)`
    background-color: ${props => props.theme.colours.form.buttonSecondary.color};
    position: fixed;
    bottom: 1rem;
    right: 1rem;
    height: 2.75rem;
    width: 2.75rem;
    z-index: 3;
    &:hover {
        background-color: ${props => props.theme.colours.form.buttonSecondary.color};
    }
`
