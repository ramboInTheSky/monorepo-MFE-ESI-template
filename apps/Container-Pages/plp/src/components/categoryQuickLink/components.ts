import styled from "styled-components"
import {Typography} from "@mui/material"
import {breakpoints} from "@monorepo/themes"

export const CategoryQuickLinkRoot = styled.a`
    width: 7.5rem;
    text-decoration: none;

    @media (min-width: ${breakpoints.values.md}px) {
        width: 11.25rem;
    }
`

export const CategoryQuickLinkImageContainer = styled.div`
    width: 7.5rem;
    height: 11.25rem;

    @media (min-width: ${breakpoints.values.md}px) {
        width: 11.25rem;
        height: 16.875rem;
    }
`

export const CategoryQuickLinkImage = styled.img`
    width: 100%;
`

export const CategoryQuickLinkTitle = styled(Typography)`
    font-size: 0.875rem;
    font-family: ${({theme}) => `${theme.colours.font.primary.regular.family}, ${theme.colours.font.default}`};
    line-height: 1.3125rem;
    letter-spacing: 0.0075rem;
    margin-top: 0.625rem;
    &:hover {
        text-decoration: underline;
    }
`

export const CategoryQuickLinkDescription = styled(Typography)`
    font-size: 0.875rem;
    font-family: ${({theme}) => `${theme.colours.font.primary.regular.family}, ${theme.colours.font.default}`};
    line-height: 1.3125rem;
    letter-spacing: 0.12;
    margin-top: 0.5rem;
`
