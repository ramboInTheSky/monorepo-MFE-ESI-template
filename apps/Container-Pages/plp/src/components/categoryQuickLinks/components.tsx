import styled from "styled-components"
import {Typography} from "@mui/material"
import {breakpoints} from "@monorepo/themes"

export const CategoryQuickLinksRoot = styled.div`
    width: 100%;
    & > :not(:first-child) {
        margin-top: 0.875rem;
    }

    /* 
        Lets ensure that the slides are not centered when there 
        are not enough slides to horizontally fill the area 
    */
    & .swiper-container {
        margin-left: 0;
        margin-right: 0;
    }

    & .swiper-wrapper {
        height: auto;
    }

    & .swiper-slide {
        width: 7.5rem;
        @media (min-width: ${breakpoints.values.md}px) {
            width: 11.25rem;
        }

        /*
            Disable the padding on the right for when there are
            not enough slides to horizontally fill the area. If
            it's not disabled, then the chevron will appear at
            side of the image and not on top of the image
        */
        &:last-child {
            margin-right: 0 !important;
        }
    }

    & .swiper-button-prev,
    .swiper-button-next {
        color: #000;
        width: 1.75rem;
        height: 2.25rem;
        background: rgba(255, 255, 255, 1);
        display: none;
        top: 38%;
        &.swiper-button-disabled {
            display: none;
        }
        &:after {
            font-size: 0.875rem;
            font-weight: bold;
        }
        @media (min-width: ${breakpoints.values.lg}px) {
            display: flex;
        }
    }
    & .swiper-button-prev {
        left: 0;
        border-left: 0;
        border-radius: 0 0.5rem 0.5rem 0;
        &:after {
            margin-left: -0.125rem;
        }
    }
    & .swiper-button-next {
        right: 0;
        border-right: 0;
        border-radius: 0.5rem 0 0 0.5rem;
        &:after {
            margin-right: -0.125rem;
        }
    }

    /* IE Specific styles */
    @media all and (-ms-high-contrast: none) {
        & .swiper-button-prev,
        .swiper-button-next {
            position: absolute;
            top: 36%;
            width: 1.6875rem;
            margin-top: -1.375rem;
            z-index: 10;
            cursor: pointer;
            background-size: 1.25rem 1rem;
            background-position: center;
            background-repeat: no-repeat;
        }
        & .swiper-button-prev.swiper-button-disabled,
        .swiper-button-next.swiper-button-disabled {
            opacity: 0.35;
            cursor: auto;
            pointer-events: none;
        }
        & .swiper-button-prev,
        .swiper-container-rtl .swiper-button-next {
            background-image: url("data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D'http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg'%20viewBox%3D'0%200%2027%2044'%3E%3Cpath%20d%3D'M0%2C22L22%2C0l2.1%2C2.1L4.2%2C22l19.9%2C19.9L22%2C44L0%2C22L0%2C22L0%2C22z'%20fill%3D'%23000000'%2F%3E%3C%2Fsvg%3E");
            left: -0.0625rem;
            right: auto;
        }
        & .swiper-button-next,
        .swiper-container-rtl .swiper-button-prev {
            background-image: url("data:image/svg+xml;charset=utf-8,%3Csvg%20xmlns%3D'http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg'%20viewBox%3D'0%200%2027%2044'%3E%3Cpath%20d%3D'M27%2C22L27%2C22L5%2C44l-2.1-2.1L22.8%2C22L2.9%2C2.1L5%2C0L27%2C22L27%2C22z'%20fill%3D'%23000000'%2F%3E%3C%2Fsvg%3E");
            right: -0.0625rem;
            left: auto;
        }
    }
`

export const CategoryQuickLinksTitle = styled(Typography).attrs({
    component: "h1",
})`
    font-size: 1rem;
    font-family: ${({theme}) => `${theme.colours.font.primary.medium.family}, ${theme.colours.font.default}`};
    font-weight: 500;
    line-height: 1.5rem;
    letter-spacing: 0.0075rem;
`

export const CategoryQuickLinkItems = styled.div`
    width: 100%;
    overflow-x: hidden;
    display: flex;
`
