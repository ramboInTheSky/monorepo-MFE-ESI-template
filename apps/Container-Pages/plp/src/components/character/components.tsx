import Link from "@mui/material/Link"
import styled from "styled-components"
import Theme from "../../models/theme"

type CharacterLinkProps = {
    disabled?: boolean
    selected?: boolean
    component: string
    theme: Theme
}

export const CharacterLink: any = styled(Link)<CharacterLinkProps>`
    && {
        padding: 0.1875rem;
        overflow-anchor: none;
        font-size: 0.875rem;
        font-family: ${(props: CharacterLinkProps) => props.theme.colours.font.primary.medium.family};
        font-weight: ${(props: CharacterLinkProps) => props.theme.colours.text.navigation.fontWeight};
        color: ${(props: CharacterLinkProps) => {
            if (props.disabled) {
                return props.theme.colours.text.disabled
            }
            return props.selected ? props.theme.colours.text.default : props.theme.colours.text.hyperlink
        }};
        cursor: ${(props: CharacterLinkProps) => (props.disabled ? "default" : "pointer")};
    }
    &&:hover {
        text-decoration: ${(props: CharacterLinkProps) => (props.disabled ? "none" : "underline")};
    }
`
