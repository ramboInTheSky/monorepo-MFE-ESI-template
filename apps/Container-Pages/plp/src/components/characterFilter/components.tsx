import styled from "styled-components"

export const StyledCharacterFilterContainer = styled.div`
    display: flex;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    padding: 1.0625rem 1.0625rem 1.0625rem 1.0625rem;
    margin: 0;
    border-bottom: ${props => props.theme.colours.plp.facetDivider};
`
