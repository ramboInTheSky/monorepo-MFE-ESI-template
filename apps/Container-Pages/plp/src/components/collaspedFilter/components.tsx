import styled from "styled-components"
import Typography from "@mui/material/Typography"

export const FacetName = styled(Typography)`
    margin: 0.5rem 0 0;
`
export const FacetList = styled.span`
    text-overflow: ellipsis;
    max-width: 210px;
    white-space: nowrap;
    overflow: hidden;
    display: block;
    font-size: 0.875rem;
    letter-spacing: 0.008rem;
    line-height: 1.5;
`
