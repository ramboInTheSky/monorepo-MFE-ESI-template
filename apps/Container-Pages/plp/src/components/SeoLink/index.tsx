import React from "react"
import {IS_BROWSER} from "../../utils/window"
import {SEOLinkComponent} from "./components"

export interface SeoLinkProps {
    url: string
    hasNextPage?: boolean
    title: string
}

export const buildUrl = ({url, hasNextPage}) => {
    const pageRegex = /(p=.*)+/g
    const pageFragment = url.match(pageRegex)?.[0] || ""
    let returnUrl = ""
    const page = parseInt(pageFragment.split("=")[1], 10)
    // going forward
    if (hasNextPage) {
        if (Number.isNaN(page)) {
            returnUrl = url.concat(`${url.includes("?") ? "&" : "?"}p=2`)
        } else {
            returnUrl = url.replace(pageRegex, `p=${page + 1}`)
        }
        // going backwards
    } else if (Number.isNaN(page) || page === 1) {
        return null
    } else {
        const prevPage = page - 1
        returnUrl = url.replace(pageRegex, prevPage === 1 ? "" : `p=${prevPage}`)
        // strip trailing ? and &
        returnUrl = returnUrl.replace(/\?$/, "")
        returnUrl = returnUrl.replace(/&$/, "")
    }

    return returnUrl
}

export const SeoLink = ({url, hasNextPage = false, title}: SeoLinkProps) => {
    if (IS_BROWSER()) return <></>

    const returnUrl = buildUrl({url, hasNextPage})
    if (!returnUrl) return <></>

    return (
        <noscript>
            <SEOLinkComponent href={returnUrl}>{title}</SEOLinkComponent>
        </noscript>
    )
}
