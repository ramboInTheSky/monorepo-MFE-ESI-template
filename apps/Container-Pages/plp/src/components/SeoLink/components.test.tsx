import React from "react"
import {render} from "@testing-library/react"
import {SEOLinkComponent} from "./components"

describe("Snapshots - products title", () => {
    test("StyledProductTitle", () => {
        const {asFragment} = render(<SEOLinkComponent />)
        expect(asFragment()).toMatchSnapshot()
    })
})
