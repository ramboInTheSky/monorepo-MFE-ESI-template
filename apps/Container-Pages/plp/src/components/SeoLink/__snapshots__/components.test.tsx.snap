// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots - products title StyledProductTitle 1`] = `
<DocumentFragment>
  .c0 {
  margin-top: 2rem;
  padding: 1rem 4rem;
  border: 1px solid black;
  color: black;
  display: block;
  list-style: none;
}

<a
    class="c0"
  />
</DocumentFragment>
`;
